internal/controller/filesystem_helpers.go

```diff
@@ -196,7 +196,9 @@ func newLvmBlockDevice(ctx context.Context, c client.Client, nnfNodeStorage *nnf
 			return nil, dwsv1alpha2.NewResourceError("could not get NnfNodeBlockStorage: %v", client.ObjectKeyFromObject(nnfNodeBlockStorage)).WithError(err).WithUserMessage("could not find storage allocation").WithMajor()
 		}
 
-		devices = nnfNodeBlockStorage.Status.Allocations[index].Accesses[os.Getenv("NNF_NODE_NAME")].DevicePaths
+		if len(nnfNodeBlockStorage.Status.Allocations) > 0 && len(nnfNodeBlockStorage.Status.Allocations[index].Accesses) > 0 {
+			devices = nnfNodeBlockStorage.Status.Allocations[index].Accesses[os.Getenv("NNF_NODE_NAME")].DevicePaths
+		}
```

Comment:
Do we need to take a different action if there are no devices? I'm not sure how to proceed in that case. I'm not exactly sure how I even got here, but this fix did allow the workflow to complete teardown instead of a CrashLoopBackoff.

Reply:
Was there a failure to allocate namespaces prior to this?

Reply:
Without the devices here, it seems like you just exit.

Reply:
So it couldn't do the teardown because it didn't have allocated devices. But in that case did all of the work in `deleteAllocation()` actually happen, maybe earlier?
Do we know whether it ever had allocated devices?

Reply:
I'm not sure in either case, I just know the pod hit this case after some behavior that probably shouldn't have happened. I'm not sure which tests ran and which didn't. Sorry I don't have more detail.

Reply:
Let's add a log.Error() here and see how things shake out in the future.